src/sentry/mediators/param.py

```diff
@@ -73,9 +77,17 @@ class Param:
         >>>     name = Param(str, default=lambda self: self.user['name'])
     """
 
-    def __init__(self, type, **kwargs):
-        self._type = type
-        self.kwargs = kwargs
+    def __init__(
+        self,
+        type: type[T],
+        *,
+        default: T | Callable[..., T] | None = None,
+        required: bool = True,
+    ) -> None:
+        self.type = type
+        self._default = default
+        self.is_required = required
+        self.has_default = default is not None
```

Comment:
this was doing a lot of memoizing -- but these are only computed once at import time anyway so I simplified things